### DISCUSSION TITLE
Microsoft Windows Camera app appium buttons are visible but not accessible using appium API's

### DISCUSSION THREAD

**Discussion by bnagalak** [appium/appium#18275] [Q&A]
Trying to automate Windows camera app using appium\selenium api's in Windows 11. we observe that after switching between video and photo mode for couple of times.. video icon will be active but **find_element_by_name("Take video")** will show as button is not present. same issue is observed with '**Take photo**' as well. Switching between photo/ video mode also not helping to resolve this issue.

Also, when we are in take video, '**Switch to video mode**' will be ideally greyed out, but appium is able to detect the element and **find_element_by_name("Take video").click()** also works but ideally do nothing on UI.

Please let us know if any one has any ideas to resolve this issue.

**Comment by portyu9** [comment]
Hi @bnagalak 

I would avoid locating the Camera control by its changing `Name`.

Windows Camera reuses the same control and changes its accessible name depending on the current mode. Locate it by its stable `AutomationId` instead:

```python
from appium.webdriver.common.appiumby import AppiumBy

button = driver.find_element(
    AppiumBy.ACCESSIBILITY_ID,
    "CaptureButton_1"
)
```

After every Photo/Video mode switch, **re-find the element** instead of keeping the previous `WebElement`, then wait for the expected state:

```python
button = driver.find_element(
    AppiumBy.ACCESSIBILITY_ID,
    "CaptureButton_1"
)

assert button.is_enabled()
button.click()
```

The fact that Appium can still find a greyed-out control is not necessarily an error—Windows UI Automation may keep disabled controls in the accessibility tree. Check `is_enabled()` rather than using element presence to determine whether that action is currently available.

If the freshly retrieved UI Automation tree itself continues exposing the wrong state after the Camera UI changes, that is a WinAppDriver/Windows Camera accessibility-tree limitation rather than a selector problem. The current Appium Windows driver still proxies Microsoft's WinAppDriver, which is no longer actively maintained; Appium now also points users to NovaWindows Driver as a modern alternative.

So the fix is: **use the stable AutomationId, reacquire the element after every mode change, and check its enabled state rather than relying on the changing Name.**

Please mark this answer as accepted if it helped, thank you.
